sklearn/tree/tree.py

```diff
@@ -269,6 +277,15 @@ def fit(self, X, y, sample_weight=None, check_input=True):
         min_samples_split = max(self.min_samples_split,
                                 2 * self.min_samples_leaf)
 
+        # If multiple trees are built on the same dataset, we only want to
+        # presort once. Splitters now can accept presorted indices if desired, 
+        # but do not handle any presorting themselves. Ensemble algorithms which
+        # desire presorting must do presorting themselves and pass that matrix
+        # into each tree.
+        if X_idx_sorted is None and self.presort == True and not issparse(X):
```

Comment:
This behavior doesn't seems to be well represented in doctstring of presort. Have I missed something?